bgpd/bgp_route.c

```diff
@@ -3449,8 +3452,12 @@ int bgp_update(struct peer *peer, struct prefix *p, uint32_t addpath_id,
 			if (pi->extra && pi->extra->bgp_orig)
 				bgp_nexthop = pi->extra->bgp_orig;
 
-			if (bgp_find_or_add_nexthop(bgp, bgp_nexthop, afi, pi,
-						    NULL, connected)
+			nh_afi = (afi == AFI_L2VPN) ?
+				 EVPN_NH_AFI_FROM_NHLEN(pi->attr) : afi;
```

Comment:
this condition checking is looking somehow weird (for me), can't `(afi == AFI_L2VPN)` be added to the same `EVPN_NH_AFI_FROM_NHLEN`  as well for readability purposes?

Reply:
Took care of this comment in the new diff.